src/fairness/weighted_tree/test/load_accounting_db.cpp

```diff
@@ -0,0 +1,302 @@
+    }
+
+    // vector of strings to hold sub banks
+    std::vector<std::string> banks;
+
+    // we've reached a bank with no sub banks, so print
+    // out all associations under this sub bank
+    if (rc != SQLITE_ROW) {
+        rc = sqlite3_bind_text (b_select_associations_stmt,
+                                1,
+                                bank_name.c_str (),
+                                -1,
+                                NULL);
+        if (rc != SQLITE_OK) {
+            fprintf(stderr, "Failed to fetch data: %s\n", sqlite3_errmsg (DB));
+            sqlite3_close (DB);
+
+            return nullptr;
+        }
+
+        // execute SQL statement
+        rc = sqlite3_step (b_select_associations_stmt);
+        while (rc == SQLITE_ROW) {
+            std::string username = reinterpret_cast<char const *> (
+                sqlite3_column_text (b_select_associations_stmt, 0));
+            std::string user_shares = reinterpret_cast<char const *> (
+                sqlite3_column_text (b_select_associations_stmt, 1));
+            std::string user_job_usage = reinterpret_cast<char const *> (
+                sqlite3_column_text (b_select_associations_stmt, 3));
+
+            // add user as a child of the node
+            user_node = std::make_shared<weighted_tree_node_t> (
+                            parent_bank,
+                            username,
+                            true,
+                            std::stoll (user_shares),
+                            std::stoll (user_job_usage));
+            node->add_child (user_node);
+            leaf_bank_usage += std::stoi (user_job_usage);
+
+            rc = sqlite3_step (b_select_associations_stmt);
+        }
+        if (rc == SQLITE_ERROR) {
+            fprintf (stderr, "%s\n", sqlite3_errmsg (DB));
+            sqlite3_close (DB);
+
+            return nullptr;
+        }
+        // aggregate usage from user nodes up to their respective banks
+        // and up to the root bank
+        for (std::shared_ptr<weighted_tree_node_t> curr_ancestor = node;
+             curr_ancestor != nullptr;
+             curr_ancestor = curr_ancestor->get_parent ()) {
+
+            curr_ancestor->set_usage (curr_ancestor->get_usage ()
+                                      + leaf_bank_usage);
+        }
+    }
+    // otherwise, this bank has sub banks, so call this
+    // function again with the first sub bank it found
+    else {
+        parent_bank = node;
+        while (rc == SQLITE_ROW) {
+            // execute SQL statement
+            std::string bank = reinterpret_cast<char const *> (
+                sqlite3_column_text (b_select_sub_banks_stmt, 0));
+            banks.push_back (bank);
+            rc = sqlite3_step (b_select_sub_banks_stmt);
+        }
+        if (rc == SQLITE_ERROR) {
+            fprintf (stderr, "%s\n", sqlite3_errmsg (DB));
+            sqlite3_close (DB);
+
+            return nullptr;
+        }
+        for (std::string b : banks) {
```

Comment:
Don't construct a new string for performance ( minor though)
```suggestion
        for (const std::string &b : banks) {
```